rules/badge.js

```diff
@@ -13,22 +14,24 @@ module.exports = (ast, file) => {
 		let hasBadge = false;
 
 		for (const child of node.children) {
-			if (node.depth === 1 && child.type === 'link' && child.url === badgeUrl) {
-				for (const child2 of child.children) {
-					if (child2.type === 'image') {
-						if (child2.url !== badgeSrcUrl) {
-							file.warn('Incorrect badge source', child2);
-							return;
-						}
+			if (node.depth !== 1 || child.type !== 'link' || child.url !== badgeUrl) {
```

Comment:
i only made this change because `xo` was complaining about too much nesting depth here